server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/features/ResetFeatureStateResponse.java

```diff
@@ -92,59 +93,106 @@ public String toString() {
      */
     public static class ResetFeatureStateStatus implements Writeable, ToXContentObject {
         private final String featureName;
-        private final String status;
+        private final Status status;
+        private final Exception exception;
 
-        public ResetFeatureStateStatus(String featureName, String status) {
+        public enum Status {
+            SUCCESS,
+            FAILURE
+        }
+
+        public static ResetFeatureStateStatus success(String featureName) {
+            return new ResetFeatureStateStatus(featureName, Status.SUCCESS, null);
+        }
+
+        public static ResetFeatureStateStatus failure(String featureName, Exception exception) {
+            return new ResetFeatureStateStatus(
+                featureName,
+                Status.FAILURE,
+                exception);
+        }
+
+        public static ResetFeatureStateStatus failure(String featureName, String exceptionMessage) {
```

Comment:
Let's make the caller provide the exception and not allow for a message to be provided without an exception